geometry/polygon.hpp

```diff
@@ -247,6 +284,29 @@ bool insidePolygon(const PolygonType &polygon
     return true;
 }
 
+template <typename T>
+bool insideMultiPolygon(const std::vector<std::vector<math::Point2_<T>>>& mpoly,
+                        const math::Point2_<T>& point)
+{
+    bool c = false;
+    for (const auto& poly : mpoly)
+    {
+        auto nvert = int(poly.size());
+        for (int i = 0, j = nvert - 1; i < nvert; j = i++)
+        {
+            if (((poly[i](1) > point(1)) != (poly[j](1) > point(1)))
+                && (point(0) < (poly[j](0) - poly[i](0))
+                                       * (point(1) - poly[i](1))
+                                       / (poly[j](1) - poly[i](1))
+                                   + poly[i](0)))
+            {
+                c = !c;
```

Comment:
What is the idea behind this algorithm?

Reply:
This is just moved from `triangulate.cpp`:
https://github.com/melowntech/libgeometry/blob/cadcb2607280a315c7fc6e71efd63881d9a32a19/geometry/triangulate.cpp#L151-L172

A docstring there mentions the following source and a "TODO" to move it to `polygon.hpp`
https://stackoverflow.com/questions/217578/how-can-i-determine-whether-a-2d-point-is-within-a-polygon
Note that it's duplicated in `triangulate.cpp` - first version that does not support multipolygons is disabled.